docs/en/observability/deploy-elastic-serverless-forwarder.asciidoc

```diff
@@ -0,0 +1,164 @@
+
+* <<aws-serverless-forwarder-deploy-console>>
+* <<aws-serverless-forwarder-deploy-cloudformation>>
+* <<aws-serverless-forwarder-deploy-terraform>>
+
+[[aws-serverless-forwarder-deploy-console]]
+=== Deploy using {aws} Console
+
+. Login to {aws} console and navigate to the Lambda service.
+. Click **Create function**.
+. Select **Browse serverless app repository**.
+. Select **Public applications**.
+. Search for **elastic-serverless-forwarder** and select it from results.
++
+[role="screenshot"]
+image::images/aws-serverless-forwarder-create-function.png[Create Elastic Serverless Forwarder Lambda function within SAR]
++
+. Complete the **Application settings** as follows:
+    * `ElasticServerlessForwarderS3ConfigFile` set the value of the S3 URL in the format `s3://bucket-name/config-file-name` to the configuration file for your deployment of Elastic Serverless Forwarder - this will populate the `S3_CONFIG_FILE` environment variable of the deployed Lambda.
+    * `ElasticServerlessForwarderSSMSecrets` add a comma delimited list of {aws} SSM Secrets ARNs referenced in the config yaml file (if any).
+    * `ElasticServerlessForwarderKMSKeys` add a comma delimited list of {aws} KMS Keys ARNs to be used for decrypting {aws} SSM Secrets referenced in the config yaml file (if any).
+    * `ElasticServerlessForwarderSQSEvents` add a comma delimited list of Direct SQS queues ARNs to set as event triggers for the Lambda (if any).
+    * `ElasticServerlessForwarderS3SQSEvents` add a comma delimited list of S3 SQS Event Notifications ARNs to set as event triggers for the Lambda (if any).
+    * `ElasticServerlessForwarderKinesisEvents` add a comma delimited list of Kinesis Data Stream ARNs to set as event triggers for the Lambda (if any).
+    * `ElasticServerlessForwarderCloudWatchLogsEvents` add a comma delimited list of Cloudwatch Logs Log Groups ARNs to set subscription filters on the Lambda (if any).
+    * `ElasticServerlessForwarderS3Buckets` add a comma delimited list of S3 buckets ARNs that are sources for the S3 SQS Event Notifications (if any).
+. Click **Deploy** once your settings have been added.
+. On the Applications page for **serverlessrepo-elastic-serverless-forwarder**, click **Deployments**.
+. Refresh the **Deployment history** until you see the `Create complete` status update.
+. (Optional) To enable Elastic APM instrumentation for your new deployment:
+    * Go to **Lambda > Functions** within {aws} console, and find and select the function with **serverlessrepo-elastic-se-ElasticServerlessForward-**.
+    * Go to **Configuration** tab and select **Environment Variables**
+    * Add the following environment variables:
+
+      | Key                       | Value  |
+      |---------------------------|--------|
+      |`ELASTIC_APM_ACTIVE`       | `true` |
+      |`ELASTIC_APM_SECRET_TOKEN` | token  |
+      |`ELASTIC_APM_SERVER_URL`	  | url    |
+
+[[aws-serverless-forwarder-deploy-cloudformation]]
```

Comment:
What happens next? Will users see the data streams in Elasticsearch? I tried this section, which should have been the easiest for me, but must have messed something up because I'm not getting any logs in Elasticsearch.

Reply:
@dedemorton 

>Could not test this step... serverlessrepo-elastic-se-ElasticServerlessForward does not exist for me under Lambda > Functions. I see this, but I'm using a shared environment.

how did you deployed the forwarder? according to the method (console, terraform, cloudformation) the prefix before "MacroElastic..." and "ApplicationElastic..." may change.
You could try to refresh the list of lambdas and order by last updated.



> What happens next? Will users see the data streams in Elasticsearch? I tried this section, which should have been the easiest for me, but must have messed something up because I'm not getting any logs in Elasticsearch.

it should. I wonder if the deployment of the forwarder was successful at all in your test.
if you go to the Cloudformation console can you see the stack? what's its status?
you should have something like this:
<img width="1380" alt="Schermata 2022-10-15 alle 09 06 08" src="https://user-images.githubusercontent.com/221530/195959630-29eaaf2b-6ad7-481e-8180-a38901873723.png">

being a shared environment, if a forwarder was already deployed, including the macro, you cannot deploy another one in the same region

Reply:
@aspacca I used the console. I found the instructions about the configuration and application settings to be pretty confusing, so I'm sure I must have configured things incorrectly.

Reply:
I had another go at getting this working today, with mixed results...I can see one function working but not see the data in Elastic yet. I also deleted a function that failed to deploy - perhaps because it is a shared environment and a forwarder was already deployed? i.e. 

> being a shared environment, if a forwarder was already deployed, including the macro, you cannot deploy another one in the same region

I think a short zoom might be useful to ask you some questions and potentially troubleshoot some deployment issues - will ping you and see if we can find a time!

Reply:
I got it all working yesterday after a zoom session with Andrea - thank you!

On this point:
> Could not test this step... serverlessrepo-elastic-se-ElasticServerlessForward does not exist for me under Lambda > Functions. I see this, but I'm using a shared environment.

I can see mine but it is named _serverlessrepo-odh-elasti-ApplicationElasticServer-X59MCXID555K_ obviously taking odh from one of my named objects. You could check and see if you see anything named serverlessrep- @dedemorton? Or when you have time we could take a look at your setup together if you're still keen to get it working (perhaps when I push a better version of the docs?)

Have updated that doc line to:

> By default this role starts with the name **serverlessrepo-**.

Reply:
>You could check and see if you see anything named serverlessrep-

Yes, I do see some functions that begin with serverlessrepo (see the screen captures that I attached above). @dedemorton?

> we could take a look at your setup together if you're still keen to get it working (perhaps when I push a better version of the docs?)

My plate is pretty full for the next couple of weeks. Maybe when I get back from PTO in January?

Reply:
Yeah, that would be fine - it's difficult to dip into _all-the-aws-stuff_ with limited time and I think we will have enough review improvements to get this PR merged in before then and review again when we look at AWS and Cloud more completely.